graphite/graphite.go

```diff
@@ -79,6 +84,37 @@ func (f *GraphitePublisher) Publish(metrics []plugin.Metric, cfg plugin.Config)
 		}
 		data := fmt.Sprintf("%v", m.Data)
 		logger.Debug("Metric ready to send %s:%s", key, data)
+
+		for _, char := range notAllowedChars {
```

Comment:
why did you loop over notAllowedChars?  Shouldn't you check for illegal chars in the metric namespace and tags?  if they exist, replacing them with legal chars?